web/src/components/auth/utils.js

```diff
@@ -0,0 +1,194 @@
+import axios from "axios";
+import {
+  apiUrlCreateJWTToken,
+  apiUrlRegister,
+  apiUrlLoadUser,
+  apiUrlVerifyToken,
+  apiUrlVerifyActivation,
+} from "../../api.util";
+
+/**
+ * Shape of User stored in database
+ * @typeof User
+ * @property {string} first_name
+ * @property {string} last_name
+ * @property {string} email
+ * @property {int} id
+ */
+
+/**
+ * Shape of JWT Web Token pair
+ * @typeof WebToken
+ * @property {string} refresh
+ * @property {string} access
+ */
+
+/**
+ * Stores access token in Local Storage
+ * @param {string} access the access token
+ */
+const setAccessTokenOnSuccess = (access) => {
+  localStorage.setItem("access", access);
+};
+
+/**
+ * Sets access token to null on failure in Local Storage
+ *
+ */
+const setTokensOnFailure = () => {
+  localStorage.removeItem("access");
+  localStorage.removeItem("refresh");
+};
+
+/**
+ * Logs-in a user
+ * @param {string} email the user's email
+ * @param {string} password the user's password
+ * @returns {boolean}
+ */
+export const login = async (email, password) => {
```

Comment:
The primary purpose of this function seems to be `make an api call to retrieve an access token.` I think it would be more maintainable to keep this function limited to that function by:

1) Changing the structure of return data to an array of two strings. The calling function can handle the responses accordingly
   - With success: return the token in the first index and an empty string in the second index
   - With Failure: return an empty string in the first index and the error in the second index
 2) Renaming the function to `getJWT` (the "T"  means token)
 3) Removing other functions that also cause side effects, unrelated to directly getting the JWT

Reply:
Sure, I can do that. 
1. What is the reason for using an array vs. an object?

Reply:
An object would work, too. Under the hood, arrays are technically objects, anyway.